ta/os_test/sub.mk

```diff
@@ -20,3 +20,11 @@ cxxflags-remove-cxx_tests.cpp-y += -pg
 srcs-y += cxx_tests_c.c
 cflags-remove-cxx_tests_c.c-y += -pg
 endif
+
+ifeq ($(sm),ta_arm64)
+srcs-$(CFG_TA_PAUTH) += pauth_a64.S
+srcs-$(CFG_TA_PAUTH) += ta_arm_pauth.c
+# -march=armv8.3-a enables the non-nops instructions for PAC. We are using one
+#  such instruction in the PAC test.
+cflags-$(CFG_TA_PAUTH) += -march=armv8.3-a
```

Comment:
Why isn't this always added to `ta_arm64-platform-cflags` when `CFG_TA_PAUTH` is `y`?

Reply:
The instructions paciasp and aut which are used by compiler when CFG_TA_PAUTH is y are nop instructions. Using them without -march=armv8.3-a doesn't return a compilation error. While the instruction PACIZA used in the testcase here is a non-nop instruction for which the flag armv8.3-a needs to be enabled. If in future we start using non-nop PAuth instruction in TEE, we can introduce the cflag.

Reply:
I don't understand why the TA itself is supposed to keep track of these flags. I'd expect the TA dev-kit to do that.

Reply:
I can add it to the ta dev-kit if you think that would be the right place. This is not required usually for TA's using PAuth. It is only required if TA's explicitly use a non-nop PAuth instruction using assembly/inline assembly.

Reply:
With -mbranch-protection , paciasp and autiasp instructions are added as needed. These instructions are treated as nops when they are executed on architectures < v8.3. So compiler doesn't complain when you use these instructions. However instruction paciza is a non-nop instruction, so if it is being used it won't work on architectures < v8.3 i.e won't be treated as nops. So compiler complains for these instructions without explicit -march flag.

Ideally it would be good to have -march flag specified, however this should be driven by the platform port.